cedar-policy-core/src/ast/policy_set.rs

```diff
@@ -169,6 +173,8 @@ impl PolicySet {
         if self.templates.contains_key(t.id()) {
             Err(PolicySetError::Occupied)
         } else {
+            // PANIC SAFETY Checked templates did not contain this key above
+            #[allow(clippy::unreachable)]
             if self.templates.insert(t.id().clone(), Arc::new(t)).is_some() {
                 unreachable!("Template key was already present");
             }
```

Comment:
can we use Entry API to avoid this (and similar) `unreachable`? And improve performance while we're at it?